src/Rxjs.res

```diff
@@ -107,6 +107,7 @@ external toObservable: t<'c, 's, 'a> => t<foreign, void, 'a> = "%identity"
 @module("rxjs") external deferPromise: (unit => Js.Promise.t<'a>) => t<foreign, void, 'a> = "defer"
 @module("rxjs") external delayWhen: (unit => t<'c, 's, ()>, . t<'c, 's, 'a>) => t<'c, 's, 'a> = "delayWhen"
 @module("rxjs") external distinct: (unit, . t<'co, 'so, 'a>) => t<'co, 'so, 'a> = "distinct"
+@module("rxjs") external distinctUntilChanged: (@uncurry (('a, 'a) => bool), . t<'co, 'so, 'a>) => t<'co, 'so, 'a> = "distinctUntilChanged"
```

Comment:
Do you have any feedback about the other uncurried function stuff in this file?  Bound to change again in the next rescript version anyway?

Reply:
The `@uncurry` operator, alas, only works for the callbacks passed as an argument. 

Somewhat unrelated but worth noting here, I played with an idea to use data-last pipe instead of RxJS pipe, something like

```rescript
let mapped = fromArray([1,2,3])
  |> map((n, _) => n+1)
  |> filter(n => n > 3)
```

While this would work in JavaScript libraries like Ramda, in ReScript this does not work because I did not find a way how to use this pipe operator with uncurried functions - for this to work, I had to change the `map` and `filter` to uncurried and the generated code then looks like this

```JavaScript
var mapped = Rxjs.filter((function (n) {
        return n > 3;
      }), Rxjs.map((function (n, param) {
            return n + 1 | 0;
          }), Rxjs.from([
              1,
              2,
              3
            ])));
```

It passes the observable into the operator function in a single call, i.e. without partial application, but unfortunately, the RxJS operators _require_ a partial application, i.e. it does not support currying and as said, I don't think you can use an uncurried function with the pipe operator other than wrapping the calls in an extra function, something like

```rescript
let mapped = fromArray([1, 2, 3])
  |> (a => map((n, _) => n + 1)(. a))
  |> (a => filter(n => n > 3)(. a))
```

Or better with placeholder

```rescript
let mapped = fromArray([1, 2, 3])
  |> map((n, _) => n + 1)(. _)
  |> filter(n => n > 3)(. _)
```

The data-last operator will be deprecated anyway, so it makes no sense to build support for that, but it was an interesting mental exercise.

One last thing, it is possible to use data-first pipe with the uncurried functions, by utilizing the placeholders as well, the result then looks identically to data-last

```rescript
let mapped = fromArray([1, 2, 3])
  ->map((n, _) => n + 1)(. _)
  ->filter(n => n > 3)(. _)
```

Pros - no need for the `pipeXY` functions
Const - harder to write and read, the `(._)` at the end is mandatory and looks weird, I'm curious if this would work with the _uncurried by default_ in ReScript v11.